src/all/unionmangas/src/eu/kanade/tachiyomi/extension/all/unionmangas/UnionMangasDto.kt

```diff
@@ -0,0 +1,170 @@
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+class UnionMangasDto(
+    val props: Props,
+    val query: QueryDto,
+) {
+    private val latestUpdateRawContent get() = props.pageProps.latestUpdateDto
+    private val popularMangaRawContent get() = props.pageProps.popularMangaDto
+    private val latestUpdateDto get() = latestUpdateRawContent?.mangas
+    private val popularMangaDto get() = popularMangaRawContent?.map { it.details }
+    private val mangaDetailsDto get() = props.pageProps.mangaDetailsDto
+
+    fun hasNextPageToLatestUpdates() = latestUpdateRawContent?.hasNextPage() ?: false
+
+    fun hasNextPageToPopularMangas() = false
+
+    fun toPopularSManga(): List<SManga> = popularMangaDto?.map(::sMangaParse) ?: emptyList()
+
+    fun toSMangaLatestUpdates(): List<SManga> = latestUpdateDto?.map(::sMangaParse) ?: emptyList()
+
+    fun toSMangaDetails() = SManga.create().apply {
+        title = mangaDetailsDto!!.title
+        genre = mangaDetailsDto!!.genres
+        thumbnail_url = mangaDetailsDto!!.thumbnailUrl
+        url = mangaUrlParse(mangaDetailsDto!!.slug, query.type)
+        status = mangaDetailsDto!!.status
+    }
+
+    private fun sMangaParse(dto: MangaDto): SManga = SManga.create().apply {
+        title = dto.title
+        thumbnail_url = dto.thumbnailUrl
+        status = dto.status
+        url = mangaUrlParse(dto.slug, query.type)
+        genre = dto.genres
+    }
+}
+
+@Serializable
+data class ChapterPageDto(
+    val totalRecode: Int = 0,
+    val currentPage: String = "0",
+    val totalPage: Int = 0,
+    @SerialName("data") val chapters: List<ChapterDto> = emptyList(),
+) {
+    fun toSChapter(langOption: LanguageOption): List<SChapter> =
+        chapters.map { chapter ->
+            SChapter.create().apply {
+                name = chapter.name
+                date_upload = chapter.date.toDate()
+                url = "/${langOption.infix}${chapter.toChapterUrl(langOption.chpPrefix)}"
+            }
+        }
+
+    private fun String.toDate(): Long =
+        try { UnionMangas.dateFormat.parse(trim())!!.time } catch (_: Exception) { 0L }
+
+    private fun ChapterDto.toChapterUrl(prefix: String) = "/${this.slugManga}/$prefix-${this.id}"
+}
+
+@Serializable
+data class ChapterDto(
+    val date: String,
+    val slug: String,
+    @SerialName("idDoc") val slugManga: String,
+    @SerialName("idDetail") val id: String,
+    @SerialName("nameChapter") val name: String,
+)
+
+@Serializable
+data class QueryDto(
+    val type: String,
+)
+
+@Serializable
+data class Props(val pageProps: PageProps)
+
+@Serializable
+data class PageProps(
+    @SerialName("data_lastuppdate") val latestUpdateDto: PageableMangaListDto?,
+    @SerialName("data_popular") val popularMangaDto: List<PopularMangaDto>?,
+    @SerialName("dataManga") val mangaDetailsDto: MangaDetailsDto?,
+    @SerialName("data") val pageListData: String?,
+)
```

Comment:
use templates to get specific dto instead of putting all of them in one as nullable

```
@Serializable
class NextData<T>(
    val props: Props<T>,
    val query: QueryDto,
)

@Serializable
class Props<T>(val pageProps: T)
```

Reply:
About it:

```json

{
    "props": {
        "pageProps": {
              "data_popular": [{}], // List<Object>
              "data_lastuppdate": {}, // Object
              "dataManga": {} // Object
        }
    }
 }
```

I can't add a custom name with ```@SerialName()``` to the ```popular_data``` list

It should be possible to add ```@SerialName()``` above other classes and use that solution, but the real problem I encountered was with ```popular_data```

```kt

@Serializable
data class PageProps(
    @SerialName("data_lastuppdate") val latestUpdateDto: MangaListDto?,
    [...]
}
```
Then
```kt

@Serializable
@SerialName("data_lastuppdate") 
class MangaListDto([...])
```

Reply:
you simply do 
```
@Serializable
class MangaListProps(
    @SerialName("data_lastuppdate") val latestUpdateDto: MangaListDto,
)
```
and use it as `NextData<MangaListProps>` in the `parseNextData` function